### PR TITLE
Feature/unit testing

### PR DESCRIPTION
**Pull request by jhoepken**
I've added more unit testing to ``test_waves.py``, provided some documentation and cleaned up along the way.

### CODE REVIEW

**Comment by ChrisBarker-NOAA** [comment]
Thanks for your interest in wave_utils!

I'll comment on the specific changes inline.

File: wave_utils/tests/test_waves.py [hunk 1]
```diff
@@ -8,28 +8,48 @@
...
     assert hasattr(waves, 'wave_number')
 
+def test_frequency():
+    """
+    Test the wave frequency
+
+    The test is done for multiple values and most importantly edge cases, due to
+    the sqrt in the equation.
+    """
+    assert abs(4.59043034 - waves.frequency(waves.g, 3.0, 0.3)) < 1e-7
```
Comment (ChrisBarker-NOAA) [inline wave_utils/tests/test_waves.py]: we should use np.isclose() for these tests.

File: wave_utils/tests/test_waves.py [hunk 2]
```diff
@@ -8,28 +8,48 @@
...
+    assert abs(0.0 - waves.frequency(waves.g, 0.0, 0.3)) < 1e-7
+    assert abs(4.59043034 - waves.frequency(waves.g, -3.0, 0.3)) < 1e-7
+
+    assert np.isnan(waves.frequency(waves.g, -3.0, 0.3)) == 0
```
Comment (ChrisBarker-NOAA) [inline wave_utils/tests/test_waves.py]: a bit better as:

`assert not np.isnan(waves.frequency(waves.g, -3.0, 0.3))`

File: wave_utils/tests/test_waves.py [hunk 3]
```diff
@@ -8,28 +8,48 @@
...
+
+def test_dispersion():
+    """
+    Test if wave dispersion is invalid for a water depth d >= 0
+    """
+    assert abs(1.199678640 - waves.dispersion(1.0)) < 1e-7
```
Comment (ChrisBarker-NOAA) [inline wave_utils/tests/test_waves.py]: np.isclose() again...

File: wave_utils/tests/test_waves.py [hunk 4]
```diff
@@ -8,28 +8,48 @@
...
+
 def test_celerity_deep():
     h = 10000 # very deep
-    omega = 2*pi/1 # short period wave
-    k = waves.wave_number(g, omega, h)   # short waves
-    C = waves.celerity(k, h, g)
-    assert C == g / omega # should be for deep water -- in this case exact!
-    
+    omega = 2*np.pi/1 # short period wave
+    k = waves.wave_number(waves.g, omega, h)   # short waves
```
Comment (ChrisBarker-NOAA) [inline wave_utils/tests/test_waves.py]: yes, this is better -- thanks!

File: wave_utils/waves.py [hunk 5]
```diff
@@ -173,132 +171,3 @@ def shoaling_coeff(omega, g, h0, h2):
         return Ks
     else:  # Deep water
         return np.sqrt((g / (2 * omega)) / Cg2)
-
-
-# if __name__ == "__main__":
```
Comment (ChrisBarker-NOAA) [inline wave_utils/waves.py]: yup -- time to get rid of all that :-)

**Review by ChrisBarker-NOAA — CHANGES_REQUESTED** [review]
this looks great, thanks!

With a little cleanup (np.isclose -- or allclose()) I'll merge.

**Comment by jhoepken** [comment]
Hi Chris,
Thanks for the remarks. I'll apply and push changes tomorrow.

**Comment by jhoepken** [comment]
Hi Chris,

I've applied the requested changes.

File: wave_utils/tests/test_waves.py [hunk 6]
```diff
@@ -8,28 +8,43 @@
...
     assert hasattr(waves, 'wave_number')
 
+def test_frequency():
+    """
+    Test the wave frequency
+
+    The test is done for multiple values and most importantly edge cases, due to
+    the sqrt in the equation.
+    """
+    assert np.isclose(4.59043034, waves.frequency(waves.g, 3.0, 0.3), atol=1e-7)
```
Comment (ChrisBarker-NOAA) [inline wave_utils/tests/test_waves.py]: generally better to use the relative tolerance -- but these are all about the same magnitude, so this is OK.
